Source/SwiftLintFramework/Rules/SwitchCaseAlignmentRule.swift

```diff
@@ -99,18 +45,124 @@ public struct SwitchCaseAlignmentRule: ASTRule, ConfigurationProviderRule {
 
         let caseLocations = caseStatements.compactMap { caseDict -> Location? in
             guard let byteOffset = caseDict.offset,
-                let (line, char) = contents.lineAndCharacter(forByteOffset: byteOffset) else {
-                    return nil
+                  let (line, char) = contents.lineAndCharacter(forByteOffset: byteOffset) else {
+                return nil
             }
+
             return Location(file: file.path, line: line, character: char)
         }
 
+        guard let firstCase = caseLocations.first,
+              let firstCaseCharacter = firstCase.character else {
+            return []
+        }
+
+        // If indent_cases is on, the first case should be indented from its containing switch.
+        if configuration.indentedCases, firstCaseCharacter <= switchCharacter {
+            return caseLocations.map(locationToViolation)
+        }
+
+        let indentation = configuration.indentedCases ? firstCaseCharacter - switchCharacter : 0
+
         return caseLocations
-            .filter { $0.character != switchCharacter }
-            .map {
-                StyleViolation(ruleDescription: type(of: self).description,
-                               severity: configuration.severity,
-                               location: $0)
-            }
+            .filter { $0.character != switchCharacter + indentation }
+            .map(locationToViolation)
+    }
+
+    private func locationToViolation(_ location: Location) -> StyleViolation {
+        return StyleViolation(ruleDescription: type(of: self).description,
```

Comment:
can we add a `reason` argument and keep the old reason if `indentedCases == false`? Then we could also use a more specific reason when it's true (e.g. `"Case statements should be indented in relation to the enclosing switch statement"`).

This would also make it easier to review oss-check violations.

Reply:
Great idea!